compiler/semantic/ast.go

```diff
@@ -13,5 +13,6 @@ import (
 // to DAG form, resolving syntax ambiguities, and performing constant propagation.
 // After semantic analysis, the DAG is ready for either optimization or compilation.
 func Analyze(ctx context.Context, seq ast.Seq, source *data.Source, head *lakeparse.Commitish) (dag.Seq, error) {
-	return semSeq(ctx, NewScope(), seq, source, head)
+	c := newChecker(ctx, source, head)
+	return c.check(seq)
```

Comment:
Existing comments and error strings almost always refer to this step as semantic analysis, so should we rename "checker" to "analyzer" for consistency with that?

Reply:
ok